libraries/adaptive-expressions/src/builtinFunctions/any.ts

```diff
@@ -0,0 +1,53 @@
+/**
+ * @module adaptive-expressions
+ */
+/**
+ * Copyright (c) Microsoft Corporation. All rights reserved.
+ * Licensed under the MIT License.
+ */
+
+import { Expression } from '../expression';
+import { ExpressionEvaluator, ValueWithError } from '../expressionEvaluator';
+import { ExpressionType } from '../expressionType';
+import { InternalFunctionUtils } from '../functionUtils.internal';
+import { MemoryInterface } from '../memory/memoryInterface';
+import { Options } from '../options';
+import { ReturnType } from '../returnType';
+
+/**
+ * Determines whether any element of a sequence satisfies a condition.
+ */
+export class Any extends ExpressionEvaluator {
+    /**
+     * Initializes a new instance of the [Any](xref:adaptive-expressions.Any) class.
+     */
+    public constructor() {
+        super(ExpressionType.Any, Any.evaluator, ReturnType.Boolean, InternalFunctionUtils.ValidateLambdaExpression);
+    }
+
+    /**
+     * @private
+     */
+    private static evaluator(expression: Expression, state: MemoryInterface, options: Options): ValueWithError {
+        let result = false;
+        const { value: instance, error: childrenError } = expression.children[0].tryEvaluate(state, options);
+        let error = childrenError;
+        if (!error) {
+            const list = InternalFunctionUtils.convertToList(instance);
+            if (!list) {
+                error = `${expression.children[0]} is not a collection or structure object to run Any`;
+            } else {
+                InternalFunctionUtils.lambdaEvaluator(expression, state, options, list, (currentItem, r, e) => {
+                    if (InternalFunctionUtils.isLogicTrue(r) && !e) {
```

Comment:
It might make sense to check the error case first so the `isLogicTrue` call is skipped.